pkg/instance/instance_integration_test.go

```diff
@@ -207,6 +207,46 @@ func TestInstanceHandler(t *testing.T) {
 		time.Sleep(3 * time.Second)
 		k8sClient.AssertPodIsNotRunning(t, deploymentInstance.GroupName, deploymentInstance.Name)
 	})
+
+	t.Run("GetPublicDeployments", func(t *testing.T) {
+		t.Log("Create deployment")
+		var deployment model.Deployment
+		body := strings.NewReader(`{
+			"name": "private-deployment",
+			"group": "group-name",
+			"description": "some description"
+		}`)
+		client.PostJSON(t, "/deployments", body, &deployment, inttest.WithAuthToken("sometoken"))
+		assert.Equal(t, "private-deployment", deployment.Name)
+		assert.Equal(t, "group-name", deployment.GroupName)
+		assert.Equal(t, "some description", deployment.Description)
+		assert.False(t, deployment.Public)
+
+		t.Log("Create public deployment")
+		body = strings.NewReader(`{
+			"name": "public-deployment",
+			"group": "group-name",
+			"description": "some description",
+			"public": true
+		}`)
+		client.PostJSON(t, "/deployments", body, &deployment, inttest.WithAuthToken("sometoken"))
+		assert.Equal(t, "public-deployment", deployment.Name)
+		assert.Equal(t, "group-name", deployment.GroupName)
+		assert.Equal(t, "some description", deployment.Description)
+		assert.True(t, deployment.Public)
+
+		t.Log("Get public deployments")
+		var groupsWithDeployments []instance.GroupsWithDeployments
+		client.GetJSON(t, "/deployments/public", &groupsWithDeployments)
+		assert.Len(t, groupsWithDeployments, 1)
```

Comment:
Using a custom struct for the assertion and https://pkg.go.dev/github.com/stretchr/testify/assert#EqualValues might make it more readable